platform/glfw/glfw_view.cpp

```diff
@@ -634,7 +636,12 @@ void GLFWView::toggle3DExtrusions(bool visible) {
     auto extrusionLayer = std::make_unique<mbgl::style::FillExtrusionLayer>("3d-buildings", "composite");
     extrusionLayer->setSourceLayer("building");
     extrusionLayer->setMinZoom(15.0f);
-    extrusionLayer->setFilter(mbgl::style::EqualsFilter { "extrude", { std::string("true") } });
+    
+    std::vector<std::unique_ptr<mbgl::style::expression::Expression>> args;
+    args.push_back(std::make_unique<mbgl::style::expression::Literal>(mbgl::style::expression::Value(std::string("extrude"))));
+    args.push_back(std::make_unique<mbgl::style::expression::Literal>(mbgl::style::expression::Value(std::string("true"))));
+    mbgl::style::expression::ParsingContext parsingContext;
+    extrusionLayer->setFilter(mbgl::style::Filter { std::move(*mbgl::style::expression::createCompoundExpression("filter-==", std::move(args), parsingContext)) });
```

Comment:
I definitely agree that creating filters (and, more generally, expressions) is very cumbersome in mbgl core.   I'm just not sure how important it is to streamline this internal API, since its primary consumer is really `ParsingContext`.

Reply:
Specifically, [this constructor](https://www.mapbox.com/android-docs/api/map-sdk/6.0.1/com/mapbox/mapboxsdk/style/expressions/Expression.html#Expression-java.lang.String-com.mapbox.mapboxsdk.style.expressions.Expression...-). There’s a similar thing on iOS/macOS with `+[NSExpression expressionWithMGLJSONObject:]` and/or the [`MGL_FUNCTION()`](https://www.mapbox.com/ios-sdk/api/4.0.0/predicates-and-expressions.html#code-mgl_function-code) format string syntax.

Reply:
I added some convenience overloads to `createCompoundExpression` and added `createLiteral` function. These reduces the boilerplate significantly:

```cpp
ParsingContext ctx;
extrusionLayer->setFilter(Filter(createCompoundExpression("filter-==", createLiteral("extrude"), createLiteral("true"), ctx)));
```